openerp-atomfeed-service/src/main/java/org/bahmni/feed/openerp/worker/OpenERPCustomerServiceEventWorker.java

```diff
@@ -67,39 +67,43 @@ private List<Parameter> getParameters(Event event) throws IOException {
 
     private List<Parameter> mapParameters(OpenMRSPatient openMRSPatient, String eventId, String feedUri, boolean isFailedEvent) {
         List<Parameter> parameters = new ArrayList<Parameter>();
-        parameters.add(createParameter("name", openMRSPatient.getName(), "string"));
-        parameters.add(createParameter("local_name", openMRSPatient.getLocalName(), "string"));
-        parameters.add(createParameter("ref", getPrimaryIdentifier(openMRSPatient).getIdentifier(), "string"));
-        parameters.add(createParameter("uuid", openMRSPatient.getUuid(), "string"));
+        addToParametersIfNotEmpty(parameters, "name", openMRSPatient.getName());
+        addToParametersIfNotEmpty(parameters, "local_name", openMRSPatient.getLocalName().isEmpty()?"-":openMRSPatient.getLocalName());
```

Comment:
The api expects "local_name" to be non-null, string that is atleast a character long. The use of "-" addresses the current issue by temporarily ensuring the "local_name" parameter is non-null and contains at least one character, allowing our sync process to proceed unimpeded. The addition of "-" serves as a temporary measure to fulfill this requirement. Once the check for "local_name" is reverted to its original state, we can safely remove the "-" added.